test/index.js

```diff
@@ -80,10 +80,35 @@ const expected = `{
 }
 `
 
-const output = prettier.format(input, {
-  filepath: 'package.json',
-  plugins: ['.'],
+;['package.json', 'foo/bar/package.json', 'foo\\bar\\package.json'].forEach(filepath => {
+  const output = prettier.format(input, {
+    filepath,
+    plugins: ['.'],
+  })
+  console.log('Testing', filepath)
+  console.assert(output === expected, 'Output does not match expected output')
+  if (output !== expected) {
+    process.exitCode = 1
+  }
+  console.log()
 })
+;['Package.json', 'package.JSON', 'package-lock.json', 'composer.json', 'package.json/composer.json'].forEach(
+  filepath => {
+    const output = prettier.format(input, {
+      filepath,
+      plugins: ['.'],
+    })
+    console.log('Testing', filepath)
+    console.assert(output === input, 'Output does not match input')
```

Comment:
I guess we can still test `foobar.json` and any other arbitrary file names if instead of `output === input` we check for `output !== expected`. WDYT?

Reply:
This plugin patches the `json-stringify` parser. As seen [here](https://github.com/prettier/prettier/blob/master/src/language-js/index.js#L45-L55), this parser is only used for a very small number of hardcoded files.

Other JSON files are parsed using the `json` parser. Adding a test for these means it would test prettier internals, not this plugin. This would be similar to adding a test for a file named `.prettierrc`, `.eslintrc`, or even `foo.js`.

Reply:
I agree with what you're saying, yet still think that it's useful to ensure the functionality of the plugin does not 'spill over' (even if looks unrealistic with the current implementation). What if something changes in Prettier itself, e.g. they decide to use `json-strigify` for all JSONs?

Reply:
A more realistic scenario is if someone runs prettier from node and explicitly passes the `parser` option. I did this in the test case now, allowing to test any filepath option.